activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -145,5 +150,9 @@ def can_be_bound?(column_name, value)
         handler_for(value).is_a?(BasicObjectHandler) &&
         !table.associated_with?(column_name)
     end
+
+    def polymorphic_array?(value)
+      value.is_a?(Array) && value.map(&:class).map { |e| e.try(:base_class) }.uniq.size > 1
```

Comment:
Sorry, didn't realize the performance implications of iterating over this array would have any significant impact.

> We should probably check if the association is actually polymorphic before this loop

Not sure what you mean by this- the purpose of the loop is to check if it is an array of polymorphic objects.  If there's a better way to do this, we can bypass the loop entirely.  How would I check if it's polymorphic before (or without) iterating through the array?